pkg/webhook/testdata/webhook.go

```diff
@@ -27,7 +27,7 @@ func (c *CronJob) SetupWebhookWithManager(mgr ctrl.Manager) error {
 		Complete()
 }
 
-// +kubebuilder:webhook:verbs=create;update,path=/validate-testdata-kubebuilder-io-v1-cronjob,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=testdata.kubebuiler.io,resources=cronjobs,versions=v1,name=validation.cronjob.testdata.kubebuilder.io,sideEffects=Some
+// +kubebuilder:webhook:verbs=create;update,path=/validate-testdata-kubebuilder-io-v1-cronjob,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=testdata.kubebuiler.io,resources=cronjobs,versions=v1,name=validation.cronjob.testdata.kubebuilder.io,sideEffects=NoneOnDryRun
```

Comment:
This change is because in `admissionregistration/v1`, the sideEffects can only be `None` and `NoneOnDryRun`

Reply:
Are you checking for this case anywhere? There should be an error if anything other than either of those two is set.

Reply:
No, there is no such check, please check this comment: https://github.com/kubernetes-sigs/controller-tools/pull/469#discussion_r468598242

Reply:
In general I would want as much static validation as possible for my operator project so that I don't have to wait for my CI's apiserver to tell me my webhook config is invalid when I upgrade from v1beta1 to v1. Is there a good reason not to statically validate side effects?

Reply:
I guess the counter-argument is that having one marker for two versions is nice, and if one value in that marker is invalid for one version then you can't generate the config for either version, which I think is what you're suggesting in your comment.

The right solution IMO is to add a `+kubebuilder:webhook:webhookVersions=<list>` field so you can specify which version to generate for a marker if you want to generate both and configure them differently. Thoughts?

Reply:
> I guess the counter-argument is that having one marker for two versions is nice, and if one value in that marker is invalid for one version then you can't generate the config for either version, which I think is what you're suggesting in your comment.

Yeah, exactly. This is my main concern so I decided not to handle the static validation and let kube-apiserver handle it, but it may not be a good solution because users have to modify the configuration every time they re-generate it.

> The right solution IMO is to add an optional +kubebuilder:webhook:webhookVersion=<v1|v1beta1> field so you can specify which version to generate for a marker if you want to generate both and configure them differently. 

I think this will work, but it will double the number of markers if the user wants to generate both `v1` and `v1beta1` webhooks. How about we do the following, with a marker `+kubebuilder:webhook:webhookVersions=<[]string>`:
- We default to generate `v1` configuration if the marker is not specified.
- Generate `v1` if the marker is `+kubebuilder:webhook:webhookVersions=v1`
- Generate `v1beta1` if the marker is `+kubebuilder:webhook:webhookVersions=v1beta1`
- Generate both `v1` and `v1beta1` if the marker is `+kubebuilder:webhook:webhookVersions=v1;v1beta1` or `+kubebuilder:webhook:webhookVersions=v1beta1;v1`, we throw error out if the `sideEffects` (later we also needed to handle `AdmissionReviewVersions`) cannot meet requirement for either versions.
In this way, we still provide a chance for the user to generate both `v1` and `v1beta1` in one marker. 

> The case where this field wasn't added to the marker is the current case, so no version list is needed and this isn't a breaking change. 

I think it will still be a breaking change because we will generate `v1` configurations by default (currently `v1beta1`)?

WDYT @estroz ?

Reply:
Works for me.

>this isn't a breaking change.

To the marker definition I mean, since current markers will work as-is.

Reply:
@estroz This has been implemented, PTAL :)